_minutes/2023-09-11-2023-09-14-tpac-extra.md

```diff
@@ -0,0 +1,520 @@
+     * [rob] Yes. Would also be nice to define the serialization method. In principle, are you okay with that?
+     * [devlin] Yes. In my view this isn't equivalent to remotely hosted code because the content script has the extension's CSP and prevents things like inline script execution.
+     * [rob] Okay great. How about CSS from the content script? In previous discussions you were favorable towards loading CSS from the content script ([issue 403](https://github.com/w3c/webextensions/issues/403)).
+     * [devlin] dom.insertCss with an arbitrary string would be fine. Avoiding file for many of these use cases is clearly beneficial. It's faster, avoids extra IPC/overhead.
+     * [rob] And to emphasize too - there is no privilege attached to code being executed. In contrast, an inline script element in Chrome bypasses the CSP of the web page to execute code. I wouldn't want the new capability to inherit privileges like this.
+     * [devlin] Wouldn't classify that as the content script having those privileges. It would inject into the main world and the main world CSP applies. Similar to that, dom.executeScript would allow the script to execute and immediately apply the CSP
+     * [rob] This could also be useful for content script managers that want to inject a snippet in another user scripts world. If there's a userscript API that synchronously returns storage values, the content script of the user script manager would maintain that storage and return values as requested. User script manager could inject the preamble to set this up and enable them to synchronously communicate.
+     * [devlin] Not sure I agree with this case. If it's in the preamble, why not modify the code you're injecting from the start?
+     * [rob] You want to get the handle to communicate across contexts. If comms are supported as part of the userScript API you don't need this. But otherwise this primitive would enable you to communicate between worlds (isolated world and user script world). In the userScripts API design, have you considered synchronous communication between user script and manager?
+     * [oliver] is the reason for that to cancel an event?
+     * [rob] There are user script APIs that are synchronous (GM_getValue), changing that would break existing scripts. Admittedly, without special API support, it would be possible to establish an initially secure communication channel between the isolated world and the user script world, by saving copies of prototypes and using a secret event name to communicate.
+     * [devlin] Having a communication channel is something we'll pursue, but I don't know that I'd comment to it being synchronous. User scripts, I hope, won't have that much data associated with them. You could asynchronously update the data exposed by the user script manager.
```

Comment:
Many popular userscripts absolutely need to run before the first JS statement of the page even if it's an inline `<script>alert(1)</script>` and they need their config synchronously at document_start.